test/test_dataloader.py

```diff
@@ -1563,6 +1563,39 @@ def test_multi_epochs_reproducibility(self):
             for batch_idx, sample in enumerate(dataloader):
                 self.assertEqual(sample.tolist(), [batch_idx % num_workers] * batch_size)
 
+    @unittest.skipIf(not IS_LINUX, "Only linux supports fork()-ing Generators.")
+    def test_worker_generator_seed(self):
+        # Tests for the RNG seeding behaviour within _worker_loop.
+        torch.manual_seed(0)
+        g1 = torch.Generator().manual_seed(1)
+        g2 = torch.Generator().manual_seed(2)
+
+        class MyDataset(SynchronizedDataset):
+            def __getitem__(self, idx):
+                self.sync_once()
+                HIGH = 100_000
+                from_global = torch.randint(0, HIGH, size=(1,)).item()
+                from_g1 = torch.randint(0, HIGH, size=(1,), generator=g1).item()
+                from_g2 = torch.randint(0, HIGH, size=(1,), generator=g2).item()
+
+                return from_global, from_g1, from_g2
+
+
+        num_workers = 2
+        dataset = MyDataset(size=10, num_workers=num_workers, batch_size=1)
+        dl = DataLoader(dataset, num_workers=num_workers)
+
+        for from_global, from_g1, from_g2 in dl:
+            # Assert RNG of all Generators are different within a given worker (each "batch" comes from a single worker)
+            assert len(set([from_global, from_g1, from_g2])) == 3
```

Comment:
Can you please remove all the plain assert from tests. You can use `self.assertEqual()`, `self.assertNot()`, etc as appropriate